etna/models/utils.py

```diff
@@ -98,3 +98,28 @@ def select_observations(
     observations = observations.loc[timestamps]
     observations.reset_index(drop=drop_timestamp, inplace=True)
     return observations
+
+
+def determine_freq(timestamps: Union[pd.Series, pd.DatetimeIndex]) -> str:
+    """
+
+    Parameters
+    ----------
+    timestamps
+        timeline to determine frequency
+
+    Returns
+    -------
+    :
+        pandas frequency string
+
+    Raises
+    ------
+    ValueError:
+        unable do determine frequency of data
+    """
+    freq = pd.infer_freq(timestamps, warn=False)
```

Comment:
Does it works the same in all the versions of pandas? In constructor of TSDataset we have try/except block for this case. May you add tests for this [here](https://github.com/tinkoff-ai/etna/blob/4b77a9941f5692a100081f236483c7ebe0bbee39/.github/workflows/test.yml#L221) -- to run the tests on different versions of pandas. May be it is also sensible to add all the tests form `datasets` forlder

Reply:
If we look closely, we raise error if function fails to determine frequency. We can use try/except just to unify error message